modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1154,7 +1154,7 @@ void GDScriptAnalyzer::resolve_function_signature(GDScriptParser::FunctionNode *
 		GDScriptParser::DataType return_type = resolve_datatype(p_function->return_type);
 		p_function->set_datatype(return_type);
 
-#ifdef DEBUG_ENABLED
+#if defined(DEBUG_ENABLED) && defined(TOOLS_ENABLED)
```

Comment:
`TOOLS_ENABLED` implies `DEBUG_ENABLED`, so no need for both. All other code that uses `default_arg_values` uses `#ifdef TOOLS_ENABLED` so there's no reason to diverge here (or all other call sites should be changed too).

The commit message and PR title should also be changed.

Reply:
Did the shortest change and renamed pr.

Feel free to change the title or design. Good night.

Reply:
The previous code used.. #ifdef ... ahhh